warehouse/query-core/src/test/java/datawave/query/iterator/TestWaitWindowObserver.java

```diff
@@ -39,7 +39,7 @@ public void start(String queryId, long yieldThresholdMs) {
     @Override
     public boolean waitWindowOverrun() {
         checksRemainingBeforeYield.decrementAndGet();
-        if (isStarted && ((randomYieldFrequency > 0 && (random.nextInt() % randomYieldFrequency == 0)) || checksRemainingBeforeYield.get() <= 0)) {
+        if (isStarted && ((randomYieldFrequency > 0 && (random.nextInt(100) < randomYieldFrequency)) || checksRemainingBeforeYield.get() <= 0)) {
```

Comment:
This looks the same as the check later. Consider extracting this out to a method call.